File: 3820-number-of-unique-xor-triplets-ii/3820-number-of-unique-xor-triplets-ii.java
class Solution {
    public int uniqueXorTriplets(int[] nums) {
        int n=nums.length;
        Arrays.sort(nums);
        long vis[]=new long[(int)1e5];
        HashSet<Long>set=new HashSet<>();
        for(int i=0;i<n;i++)for(int j=i;j<n;j++)set.add((long)nums[i]^nums[j]);
            
        int cnt=0;
        for(long it:set){
            for(int nt:nums){
                if(vis[(int)it^nt]==0)cnt++;
                vis[(int)it^nt]=1;
            }
        }

        return cnt;
    }
}

File: 3820-number-of-unique-xor-triplets-ii/3820-number-of-unique-xor-triplets-ii.cpp
class Solution {
public:
    int uniqueXorTriplets(vector<int>& nums) {
        unordered_set<int>st,s;
        int n = nums.size();
        if(n<3) return n;
        for(int i=0;i<n-1;i++){
            for(int j=i+1;j<n;j++){
                st.insert(nums[i]^nums[j]);
            }
        }
        for(auto& num:st){
            for(auto& x: nums){
                s.insert(num^x);
            }
        }
        return s.size();
    }
};